DigitalLearningSolutions.Data/DataServices/CentresDataService.cs

```diff
@@ -1,122 +1,150 @@
+                    $"No centre found for centre id {centreId}"
+                );
+            }
+
+            return name;
+        }
+
+        public IEnumerable<(int, string)> GetActiveCentresAlphabetical()
+        {
+            var centres = connection.Query<(int, string)>(
+                @"SELECT CentreID, CentreName
+                        FROM Centres
+                        WHERE Active = 1
+                        ORDER BY CentreName"
+            );
+            return centres;
+        }
+
+        public Centre? GetCentreDetailsById(int centreId)
+        {
+            var centre = connection.QueryFirstOrDefault<Centre>(
+                @"SELECT c.CentreID,
+                            c.CentreName,
+                            c.RegionID,
+                            r.RegionName,
+                            c.NotifyEmail,
+                            c.BannerText,
+                            c.SignatureImage,
+                            c.CentreLogo,
+                            c.ContactForename,
+                            c.ContactSurname,
+                            c.ContactEmail,
+                            c.ContactTelephone,
+                            c.pwTelephone AS CentreTelephone,
+                            c.pwEmail AS CentreEmail,
+                            c.pwPostCode AS CentrePostcode,
+                            c.pwHours AS OpeningHours,
+                            c.pwWebURL AS CentreWebAddress,
+                            c.pwTrustsCovered AS OrganisationsCovered,
+                            c.pwTrainingLocations AS TrainingVenues,
+                            c.pwGeneralInfo AS OtherInformation
+                        FROM Centres AS c
+                        INNER JOIN Regions AS r ON r.RegionID = c.RegionID
+                        WHERE CentreID = @centreId",
+                new { centreId }
+            );
+
+            if (centre == null)
+            {
+                logger.LogWarning($"No centre found for centre id {centreId}");
+                return null;
+            }
+
+            if (centre.CentreLogo?.Length < 10)
+            {
+                centre.CentreLogo = null;
+            }
+
+            return centre;
+        }
+
+        public void UpdateCentreManagerDetails
+        (
+            int centreId,
+            string firstName,
+            string lastName,
+            string email,
+            string? telephone
+        )
+        {
+            connection.Execute(
+                @"UPDATE Centres SET
+                    ContactForename = @firstName,
+                    ContactSurname = @lastName,
+                    ContactEmail = @email,
+                    ContactTelephone = @telephone
+                WHERE CentreId = @centreId",
+                new {firstName, lastName, email, telephone, centreId}
+            );
+        }
+        
+        public (string firstName, string lastName, string email) GetContactInfo(int centreId)
```

Comment:
Not your change, but I wonder if we want to rename this method to "GetCentreManagerDetails" to be consistent with the update?

Reply:
I think that would be a good change. But I could also remove this, and make the calls to this use GetCentreDetailsById instead, but that would retrieve more data from the DB that wouldn't get used. Thoughts?

Reply:
I think GetCentreManagerDetails is good for now to avoid too big of a refactor. 👍